torchgeo/datasets/eurocrops.py

```diff
@@ -0,0 +1,285 @@
+            res: resolution of the dataset in units of CRS
+            transforms: a function/transform that takes an input sample
+                and returns a transformed version
+            download: if True, download dataset and store it in the root directory
+            checksum: if True, check the MD5 of the downloaded files (may be slow)
+            hcat_level: use labels at this level of the
+                `HCATv2 hierarchy <https://www.eurocrops.tum.de/taxonomy.html>`__.
+                Valid levels are 3, 4, 5, and 6.
+
+        Raises:
+            DatasetNotFoundError: If dataset is not found and *download* is False.
+
+        .. versionchanged:: 0.5
+           *root* was renamed to *paths*.
+        """
+        self.paths = paths
+        self.checksum = checksum
+        assert hcat_level in [3, 4, 5, 6]
+        self.hcat_level = hcat_level
+
+        if download:
+            self._download()
+
+        if not self._check_integrity():
+            raise DatasetNotFoundError(self)
+
+        self._load_classes()
+        self.cmap = torch.zeros((len(self.classes) + 1, 4), dtype=torch.uint8)
+        for class_index in self.classes.values():
+            color = self.colors[class_index % len(self.colors)]
+            self.cmap[class_index, :] = torch.tensor(color)
+
+        super().__init__(
+            paths=paths,
+            crs=crs,
+            res=res,
+            transforms=transforms,
+            label_fn=self._get_class_index,
+        )
+
+    def _check_integrity(self) -> bool:
+        """Check integrity of dataset.
+
+        Returns:
+            True if dataset files are found and/or MD5s match, else False
+        """
+        assert isinstance(self.paths, str)
+
+        filepath = os.path.join(self.paths, self.hcat_fname)
+        if not check_integrity(filepath, self.hcat_md5 if self.checksum else None):
+            return False
+
+        for fname, year, md5 in self.zenodo_files:
+            filepath = os.path.join(self.paths, fname)
+            if not check_integrity(filepath, md5 if self.checksum else None):
+                return False
+        return True
+
+    def _download(self) -> None:
+        """Download the dataset and extract it."""
+        if self._check_integrity():
+            print("Files already downloaded and verified")
+            return
+        assert isinstance(self.paths, str)
+        download_url(
+            self.base_url + self.hcat_fname,
+            self.paths,
+            md5=self.hcat_md5 if self.checksum else None,
+        )
+        for fname, year, md5 in self.zenodo_files:
+            download_and_extract_archive(
+                self.base_url + fname, self.paths, md5=md5 if self.checksum else None
+            )
+
+    def _load_classes(self) -> None:
+        """Load classes from the HCAT CSV file.
+
+        The classes are loaded corresponding to the hcat_level specified by the user.
+        """
+        assert isinstance(self.paths, str)
```

Comment:
Seems like `self.paths` can never be a list of paths, is that correct? Not sure if it's possible to relax this constraint or if we should just change the type of the initial parameter.

Reply:
Yes it is similar to `cdl.py` I think.